### DISCUSSION TITLE
Preview Link for RStudio Download Page Hangs

### DISCUSSION THREAD

**Discussion by jmgirard** [quarto-dev/quarto-cli#1449] [Q&A] [answered]
When I try to preview a link to the RStudio download page in a RevealJS presentation, it just hangs forever.

    ---
    format: 
      revealjs:
        preview-links: true
    ---
    
    Works: [Download R](https://cloud.r-project.org)
    
    Doesn't: [Download RStudio](https://www.rstudio.com/products/rstudio/download/)

**Comment by cscheid** [comment, marked as answer]
This will generally happen to any domain which disallows embedding in iframes; RStudio is one of them. IIRC, some google pages also disallow iframe embedding.

**Reply by jmgirard** [reply]
If it's possible to detect this, it might be helpful to show a message in the preview window explaining that.

**Reply by jmgirard** [reply]
And maybe in those cases, also include a clickable link that will open in a new window.

**Comment by jjallaire** [comment]
Detecting is a bit tricky b/c we can't sniff what happens inside the iframe. However, if you see this for a given link you can opt-out for just that link as described here https://quarto.org/docs/presentations/revealjs/presenting.html#preview-links
